src/Featurizers/UnitTests/TfidfVectorizerFeaturizer_UnitTests.cpp

```diff
@@ -0,0 +1,653 @@
+// ----------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License
+// ----------------------------------------------------------------------
+#define CATCH_CONFIG_MAIN
+#include "catch.hpp"
+
+#include "../../3rdParty/optional.h"
+#include "../../Archive.h"
+#include "../TfidfVectorizerFeaturizer.h"
+#include "../Structs.h"
+#include "../TestHelpers.h"
+#include "../../Traits.h"
+
+namespace NS = Microsoft::Featurizer;
+
+using IndexMap = typename NS::Featurizers::TfidfVectorizerTransformer::IndexMap;
+using AnalyzerMethod = NS::Featurizers::Components::Details::DocumentStatisticsTrainingOnlyPolicy::AnalyzerMethod;
+using NormMethod = typename NS::Featurizers::TfidfVectorizerTransformer::NormMethod;
+
+template<typename T>
+void SparseVectorNumericCheck(NS::Featurizers::SparseVectorEncoding<T> const &a,
+                              NS::Featurizers::SparseVectorEncoding<T> const &b) {
+    CHECK(a.NumElements == b.NumElements);
+    assert(a.Values.size() == b.Values.size());
+    std::vector<NS::Featurizers::SparseVectorEncoding<std::float_t>::ValueEncoding> const & aValues(a.Values);
+    std::vector<NS::Featurizers::SparseVectorEncoding<std::float_t>::ValueEncoding> const & bValues(b.Values);
+    for (size_t idx = 0; idx < a.Values.size(); ++idx) {
+        CHECK(aValues[idx].Index == bValues[idx].Index);
+        CHECK(Approx(aValues[idx].Value) == bValues[idx].Value);
+    }
+}
+
+// for (auto const & item : result[0].Values)
+//     std::cout << item.Value << ", "<< item.Index << std::endl;
+
+TEST_CASE("string_standard_1_with_decorator") {
+    using InputType       = std::string;
+    using TransformedType = NS::Featurizers::SparseVectorEncoding<std::float_t>;
+
+    auto trainingBatches = 	NS::TestHelpers::make_vector<std::vector<InputType>>(
+                                NS::TestHelpers::make_vector<InputType>("this is THE first document"),
+                                NS::TestHelpers::make_vector<InputType>("this DOCUMENT is the second document"),
+                                NS::TestHelpers::make_vector<InputType>("and this is the THIRD one"),
+                                NS::TestHelpers::make_vector<InputType>("IS this THE first document")
+                            );
+
+    auto inferencingInput =  NS::TestHelpers::make_vector<InputType>("THIS is the FIRST document");
+
+    std::vector<TransformedType::ValueEncoding> values;
```

Comment:
>values; [](start = 48, length = 7)

{} initialization?

Reply:
sure, btw, as this PR is not target to master, will close and initialize a new one with your comments resolved